test/net.test.js

```diff
@@ -1,184 +1,157 @@
-const net = require('../src/net.js')
 const assert = require('assert');
 const test = require('mocha').it;
 const sinon = require('sinon');
 const proxyquire = require("proxyquire");
 
 describe("Calculate net salary", function () {
+    let insurancesCalcStub;
+    let taxCalcStub;
+
+    let net;
+
+    beforeEach(function () {
+        insurancesCalcStub = sinon.stub();
+        taxCalcStub = sinon.stub()
+
+        net = proxyquire("../src/net.js", {
+            "./insurancesCalculator.js": insurancesCalcStub,
+            "./taxCalculator.js": taxCalcStub,
+        });
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    })
 
     describe('Check Dependent deduction', function () {
         //Personal Income Tax (PIT)
+        const mockedInsurance = { total: 0, insurances: [] };
+        const mockedTaxes = { totalTax: 0, rates: [] };
+        beforeEach(function () {
+            insurancesCalcStub.returns(Promise.resolve(mockedInsurance));
+            // insurancesCalcStub.resolves(mockedInsurance);
+
+            taxCalcStub.returns(
+                new Promise(function (resolve, reject) {
+                    resolve(mockedTaxes);
+                })
+            )
+        })
 
-        test('No dependent deducted', function () {
+        afterEach(function () {
+            sinon.restore();
+        })
+
+        test('No dependent deducted', async function () {
             //Gross income below 11 million is tax-exempt.
-            const actualPayslip = net(12_290_500, 0);
+            const gross = 12_290_500;
+            taxCalcStub.withArgs(0).resolves(mockedTaxes);
```

Comment:
taxCalcStub is stubbed in beforeEach block. We can use that stub.

In this test case, we need to verify that the stub is called with expected args. Something like:
`assert(callback.calledWith('arg1', 'arg2'))`

Check the rest of the test cases if we need to verify the tax stub and insurance stub